### ISSUE TITLE
Allow negations on <when> tags.

### ISSUE THREAD

**Issue by gilberto-torrezan** [jefalbino/jsmart-web#1]
Consider this scenario: I have a collection of components that should be rendered only when the user doesn't have some authorization role. Currently I have to create an empty `<when>` tag with a `<otherwise>`. For example:

``` xml
<sm:authorize>
    <sm:when access="role_that_the_user_shouldnt_have"/>
    <sm:otherwise>
        <div id="my-div" />
    </sm:otherwise>
</sm:authorize>
```

And that is worse when I want to render the same `<div>` only when the user doesn't have a role AND have another one:

``` xml
<sm:authorize>
    <sm:when access="role_that_the_user_shouldnt_have"/>
    <sm:otherwise>
        <sm:authorize>
            <sm:when access="role_that_the_user_should_have">
                <div id="my-div" />
            </sm:when>
        </sm:authorize>
    </sm:otherwise>
</sm:authorize>
```

My suggestion is to have a way to add a negation to the `<when>` tag, such as:

``` xml
<sm:when access="!role_that_the_user_shouldnt_have, role_that_the_user_should_have">
```

or a new attribute, such as:

``` xml
<sm:when access="role_that_the_user_should_have" deny="role_that_the_user_shouldnt_have">
```

**Comment by jefalbino** [comment]
Good suggestion to improve the component usability. We are going to integrate the solution on 2.0.1_SNAPSHOT.

**Comment by jefalbino** [comment]
Included `grant` and `deny` attributes for `when` tag, so in case deny contains roles the `when` tag content will be executed in case current logged user has all roles denied present on `deny` attributes. 
For `grant` attribute that works as same as `access` attribute, the `when` tag content will be executed in case the current logged used has at least one role  present on `grant` attribute. Also the deny has precedence over grant roles.
The changes are available on 2.0.1_SNAPSHOT and will soon be released.
